app/controllers/underpayments/UnderpaymentStartController.scala

```diff
@@ -42,7 +43,12 @@ class UnderpaymentStartController @Inject() (
     if (request.userAnswers.get(UnderpaymentDetailSummaryPage).getOrElse(Seq.empty).nonEmpty) {
       Future.successful(Redirect(controllers.underpayments.routes.UnderpaymentDetailSummaryController.onLoad()))
     } else {
-      Future.successful(Ok(view(backLink(), request.isOneEntry, !request.checkMode)))
+      val nameOfImporterOrRep = if (request.isRepFlow) {
+        request.userAnswers.get(ImporterNamePage).get
```

Comment:
Even though the ImporterName should be present here can to retrieve it in a safe way that allows us to use the ErrorHandler if the data has been lost. Same for KnownEoriDetails below.